sdk/src/utils/xmp_inmemory_utils.rs

```diff
@@ -177,6 +198,8 @@ fn add_xmp_key(xmp: &str, key: &str, value: &str) -> Result<String> {
             }
         }
     }
+    let padding_length = target_length.saturating_sub(writer.get_ref().get_ref().len());
```

Comment:
Shouldn't this take into account the existing padding to keep the length the same if possible?  This seems like it would not take into account the existing padding.  Is there somewhere you account to the existing padding?

Reply:
Yes. Line 107
 
`target_length = orig_length - xpacket_end_length;`